src/basic/btrfs-util.c

```
@@ -1689,8 +1689,7 @@ int btrfs_subvol_snapshot_fd(int old_fd, const char *new_path, BtrfsSnapshotFlag
                 r = btrfs_subvol_make(new_path);
                 if (r == -ENOTTY && (flags & BTRFS_SNAPSHOT_FALLBACK_DIRECTORY)) {
                         /* If the destination doesn't support subvolumes, then use a plain directory, if that's requested. */
-                        if (mkdir(new_path, 0755) < 0)
-                                return r;
+                        (void) mkdir(new_path, 0755);
```

~~Ignoring all error here seems not good. I think only `EEXIST` should be ignored, if I correctly understand the issue #8780. I propose the following:~~
```
if (mkdir(new_path, 0755) < 0 && errno != EEXIST)
        return -errno;
```